### PR TITLE
fix: Dapr runtime panic when handle Pub/Sub (dapr#3281)

### PR DESCRIPTION
**Pull request by beiwei30**
# Description

fix issue dapr/dapr#3281 (Dapr runtime panic when handle Pub/Sub without CloudEvents status retry or drop #3281)

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: https://github.com/dapr/dapr/issues/3281

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pubsub/envelope.go [hunk 1]
```diff
@@ -103,6 +103,27 @@ func FromCloudEvent(cloudEvent []byte, topic, pubsub, traceID string) (map[strin
 	m[TopicField] = topic
 	m[PubsubField] = pubsub
 
+	// specify default value if it's unspecified from the original CloudEvent
+	if m[IDField] == nil {
```
Comment (pkedy) [inline pubsub/envelope.go:107]: dapr/dapr#3356 fixes the issue in the runtime. I would still have this change, but I think we should remove the `""` defaults for ID and Type and keep the rest. IMO `nil`/not set is better than an empty string when the field has really has no value.

**Review by pkedy — CHANGES_REQUESTED** [review]
I like the defaults, but not empty string for ID and Type.

File: pubsub/envelope.go (hunk 1 above)
Comment (beiwei30) [reply pubsub/envelope.go]: @pkedy pls. take a look again.

**Comment by beiwei30** [comment]
@pkedy any update on this?

**Comment by yaron2** [comment]
Thanks for this @beiwei30

**Pull request merged**